File: client/src/features/products/dto/response/product.response.ts
import type { UserBriefResponse } from '../../../../shared/dto/response'

export type ProductStatus = 'ACTIVE' | 'INACTIVE'

export interface ProductResponse {
  id: string
  tenantId: string
  name: string
  status: ProductStatus
  createdBy: UserBriefResponse
  updatedBy: UserBriefResponse
  createdAt: string
  updatedAt: string
}

File: backend/sentinel-api/src/main/java/com/sentinel/api/product/dto/response/ProductResponse.java
package com.sentinel.api.product.dto.response;

import com.sentinel.api.common.dto.response.UserBriefResponse;
import com.sentinel.api.product.entity.ProductStatus;

import java.time.Instant;

public record ProductResponse(
    String id,
    String tenantId,
    String name,
    ProductStatus status,
    UserBriefResponse createdBy,
    UserBriefResponse updatedBy,
    Instant createdAt,
    Instant updatedAt) {
}
